### PR TITLE
[WNMGDS-2623] Add a label migration guide

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

Add a migration guide for labels.

## How to test

1. `yarn start`
2. [localhost page](http://localhost:8000/migration-guides/v9-label-updates/)

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### If this is a change to documentation:

- [x] Checked for spelling and grammatical errors

### CODE REVIEW

File: packages/docs/content/migration-guides/v9-label-updates.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+title: Separating labels, hints, and errors in v9
+order: 60
```
Comment (zarahzachz) [inline packages/docs/content/migration-guides/v9-label-updates.mdx:3]: how are we ordering our migration guides? i'm looking at the list of guides we currently have and there's seemingly no system. it's not alphabetical and not chronological so i'm curious

File: packages/docs/content/migration-guides/v9-label-updates.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+intro: Labels, hints, and error messages are now in separate elements for better accessibility and flexibility
```
Comment (zarahzachz) [inline packages/docs/content/migration-guides/v9-label-updates.mdx:4]: needs to end with a period

File: packages/docs/content/migration-guides/v9-label-updates.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+## What changed and why
+
+Before this change, hints and error messages were being rendered inside a form field's `<label>` element. One exception was for fields with bottom-placed errors, where the error message would go below the field in its own element and linked to the field via `aria-describedby`. Here are some problems we found with the strategy of putting everything inside `<label>`:
+
+1. Links and other interactive content in the hints and error messages would interfere with assistive technology by being inside the `<label>`.
+2. Hints and error messages were harder to customize and style being inside the `<label>` element and its React component.
+3. Error-message placement being variable but rendered by the [Label](/components/label) component (previously called `FormLabel`) made the code harder to reason about, and that difficulty extended to custom fields made by application teams.
+4. Error-message placement being variable and implemented differently in the markup lead to inconsistent screen-reader experiences
```
Comment (zarahzachz) [inline packages/docs/content/migration-guides/v9-label-updates.mdx:14]: needs to end with a period

File: packages/docs/content/migration-guides/v9-label-updates.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  <p id="text-field--1__error" class="ds-c-inline-error">Example error message</p>
+  <input id="text-field--1" aria-describedby="text-field--1__error text-field--1__hint" />
+</div>
+```
+
+## Do I need to update my code?
+
+You don't need to change anything unless your application does any of the following:
+
+1. Creates custom form fields by either using `<FormLabel>` directly or by writing your own label markup
```
Comment (zarahzachz) [inline packages/docs/content/migration-guides/v9-label-updates.mdx:52]: i think our guidance is list items need to end with a period.

**Review by zarahzachz — APPROVED** [review]
left some comments but i don't think they block approval

File: packages/docs/content/migration-guides/v9-label-updates.mdx (hunk 4 above)
Comment (pwolfert) [reply packages/docs/content/migration-guides/v9-label-updates.mdx:52]: Ha ha, I was actually hoping you wouldn't catch that. I don't like the periods here, but I'll add them for the sake of compliance 👍

File: packages/docs/content/migration-guides/v9-label-updates.mdx (hunk 1 above)
Comment (pwolfert) [reply packages/docs/content/migration-guides/v9-label-updates.mdx:3]: That's a great question...

File: packages/docs/content/migration-guides/v9-label-updates.mdx (hunk 4 above)
Comment (pwolfert) [reply packages/docs/content/migration-guides/v9-label-updates.mdx:52]: Oh, it's in the list above this one on lines 52 and 53 that I don't like them. But I'm adding them

File: packages/docs/content/migration-guides/v9-label-updates.mdx (hunk 1 above)
Comment (pwolfert) [reply packages/docs/content/migration-guides/v9-label-updates.mdx:3]: I'll order it by date for now

**Pull request merged**